src/components/views/messages/RoomPredecessorTile.tsx

```diff
@@ -112,4 +152,41 @@ export const RoomPredecessorTile: React.FC<IProps> = ({ mxEvent, timestamp }) =>
             timestamp={timestamp}
         />
     );
+
+    function createLinkWithRoom(room: Room, roomId: string, eventId?: string): string {
+        const permalinkCreator = new RoomPermalinkCreator(room, roomId);
+        permalinkCreator.load();
+        if (eventId) {
+            return permalinkCreator.forEvent(eventId);
+        } else {
+            return permalinkCreator.forRoom();
+        }
+    }
+
+    function createLinkWithoutRoom(roomId: string, viaServers: string[], eventId?: string): string {
+        const matrixToPermalinkConstructor = new MatrixToPermalinkConstructor();
+        if (eventId) {
+            return matrixToPermalinkConstructor.forEvent(roomId, eventId, viaServers);
+        } else {
+            return matrixToPermalinkConstructor.forRoom(roomId, viaServers);
+        }
+    }
+
+    /**
+     * Guess the permalink for a room based on its room ID.
+     *
+     * The spec says that Room IDs are opaque [1] so this can only ever be a
+     * guess. There is no guarantee that this room exists on this server.
+     *
+     * [1] https://spec.matrix.org/v1.5/appendices/#room-ids-and-event-ids
+     */
+    function guessLinkForRoomId(roomId: string): string | null {
+        const m = roomId.match(/:([^:]*)$/);
```

Comment:
Can we use `const [localPart, serverName] = mxid.split(':');`?

This is what we [used in the Gitter webapp](https://gitlab.com/gitterHQ/webapp/-/blob/d89bc6c0f925cf8a79c7435e55f0b354bcc0cc75/modules/matrix-bridge/lib/parse-gitter-mxid.js#L14) (although we're really only concerned with a few hosts) and I see it [done in the `matrix-js-sdk`](https://github.com/matrix-org/matrix-js-sdk/blob/c482a6ab15f4cd352ed24761ee60ffc761e707cf/src/models/invites-ignorer.ts#L165-L166)

Reply:
this would break server names which contain ports which is not that uncommon, ipv6 a bit less common. `.replace(/^.*?:/, "")` is also done in the js-sdk. 

https://github.com/matrix-org/matrix-js-sdk/blob/develop/src/client.ts#L1799

Reply:
I went for `/[^:]*:(.*)/`. Split and join feels sad to me :-)

Reply:
I also added tests (!)